android/src/test/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapperTests.kt

```diff
@@ -0,0 +1,129 @@
+            assertThat(it["description"]).isEqualTo(expected)
+        }
+    }
+
+    @Test
+    fun `maps product title correctly`() {
+        val expected = "Expected Title"
+        stubStoreProduct(title = expected).map().let {
+            assertThat(it["title"]).isEqualTo(expected)
+        }
+    }
+
+    @Test
+    fun `maps product price correctly`() {
+        val expected = 2.0
+        stubStoreProduct(priceAmountMicros = (expected * 1_000_000).toLong()).map().let {
+            assertThat(it["price"]).isEqualTo(expected)
+        }
+    }
+
+    @Test
+    fun `maps product price string correctly`() {
+        val expected = "$2.00"
+        stubStoreProduct(price = expected).map().let {
+            assertThat(it["priceString"]).isEqualTo(expected)
+        }
+    }
+
+    @Test
+    fun `maps currency code correctly`() {
+        val expected = "CAD"
+        stubStoreProduct(priceCurrencyCode = expected).map().let {
+            assertThat(it["currencyCode"]).isEqualTo(expected)
+        }
+    }
+
+    @Test
+    fun `maps discounts correctly`() {
+        stubStoreProduct().map().let {
+            assertThat(it["discounts"]).isEqualTo(null)
+        }
+    }
+
+    @Test
+    fun `maps product category correctly`() {
+        stubStoreProduct(type = ProductType.SUBS).map().let {
+            assertThat(it["productCategory"]).isEqualTo("SUBSCRIPTION")
+        }
+        stubStoreProduct(type = ProductType.INAPP).map().let {
+            assertThat(it["productCategory"]).isEqualTo("NON_SUBSCRIPTION")
+        }
+        stubStoreProduct(type = ProductType.UNKNOWN).map().let {
+            assertThat(it["productCategory"]).isEqualTo("UNKNOWN")
+        }
+    }
+
+    @Test
+    fun `maps product type correctly`() {
+        stubStoreProduct(type = ProductType.SUBS).map().let {
+            assertThat(it["productType"]).isEqualTo("AUTO_RENEWABLE_SUBSCRIPTION")
+        }
+        stubStoreProduct(type = ProductType.INAPP).map().let {
+            assertThat(it["productType"]).isEqualTo("CONSUMABLE")
+        }
+        stubStoreProduct(type = ProductType.UNKNOWN).map().let {
+            assertThat(it["productType"]).isEqualTo("UNKNOWN")
+        }
+    }
+}
+
+fun stubStoreProduct(
+    sku: String = "monthly_product",
+    type: ProductType = ProductType.SUBS,
+    price: String = "$1.00",
+    priceAmountMicros: Long = 1_000_000,
+    priceCurrencyCode: String = "USD",
+    originalPrice: String? = "$1.00",
+    originalPriceAmountMicros: Long = 0,
+    title: String = "A product title",
+    description: String = "A product description",
+    subscriptionPeriod: String? = "P1M",
+    freeTrialPeriod: String? = "P7D",
+    introductoryPrice: String? = null,
+    introductoryPriceAmountMicros: Long = 0,
+    introductoryPricePeriod: String? = null,
+    introductoryPriceCycles: Int = 0,
+    iconUrl: String = "http://url.com",
+    originalJson: JSONObject = JSONObject("{}")
+) = StoreProduct(
```

Comment:
I guess once we merge the BC5 changes where this is an interface we will need to change this, but this works for now 👍